default/python/AI/ResourcesAI.py

```diff
@@ -400,37 +400,73 @@ def assess_protection_focus(pid, pinfo):
 
 def set_planet_growth_specials(focus_manager):
     """set resource foci of planets with potentially useful growth factors. Remove planets from list of candidates."""
-    if useGrowth:
-        # TODO: also consider potential future benefit re currently unpopulated planets
-        for metab, metab_inc_pop in ColonisationAI.empire_metabolisms.items():
-            for special in [aspec for aspec in AIDependencies.metabolismBoostMap.get(metab, []) if aspec in ColonisationAI.available_growth_specials]:
-                ranked_planets = []
-                for pid in ColonisationAI.available_growth_specials[special]:
-                    pinfo = focus_manager.all_planet_info[pid]
-                    planet = pinfo.planet
-                    pop = planet.currentMeterValue(fo.meterType.population)
-                    if (pop > metab_inc_pop - 2 * planet.size) or (GROWTH not in planet.availableFoci):  # not enough benefit to lose local production, or can't put growth focus here
-                        continue
-                    for special2 in ["COMPUTRONIUM_SPECIAL"]:
-                        if special2 in planet.specials:
-                            break
-                    else:  # didn't have any specials that would override interest in growth special
-                        print "Considering Growth Focus for %s (%d) with special %s; planet has pop %.1f and %s metabolism incremental pop is %.1f" % (
-                            planet.name, pid, special, pop, metab, metab_inc_pop)
-                        if pinfo.current_focus == GROWTH:
-                            pop -= 4  # discourage changing current focus to minimize focus-changing penalties
-                        ranked_planets.append((pop, pid, pinfo.current_focus))
-                if not ranked_planets:
-                    continue
-                ranked_planets.sort()
-                print "Considering Growth Focus choice for special %s; possible planet pop, id pairs are %s" % (metab, ranked_planets)
-                for _spPop, spPID, cur_focus in ranked_planets:  # index 0 should be able to set focus, but just in case...
-                    planet = focus_manager.all_planet_info[spPID].planet
-                    if focus_manager.bake_future_focus(spPID, GROWTH):
-                        print "%s focus of planet %s (%d) at Growth Focus" % (["set", "left"][cur_focus == GROWTH], planet.name, spPID)
-                        break
-                    else:
-                        print "failed setting focus of planet %s (%d) at Growth Focus; focus left at %s" % (planet.name, spPID, planet.focus)
+    if not useGrowth:
+        return
+
+    # TODO: also consider potential future benefit re currently unpopulated planets
```

Comment:
I suspect this TODO was added in a moment of exuberance, it no longer seems that sensible to me (having the colonization valuation calculations take into account the availability of a growith special seems to make a lot more sense than the other way around).  I'd suggest we go ahead and delete this TODO now.